src/webgpu/api/operation/memory_sync/buffer/buffer_sync_test.ts

```diff
@@ -0,0 +1,242 @@
+import { assert } from '../../../../../common/framework/util/util.js';
+import { GPUTest } from '../../../../gpu_test.js';
+
+const SIZE = 4;
+export class BufferSyncTest extends GPUTest {
+  // Create a buffer, and initiate it to a specified value for all elements.
+  async createBufferWithValue(initValue: number): Promise<GPUBuffer> {
+    const fence = this.queue.createFence();
+    this.queue.signal(fence, 1);
```

Comment:
The fence is signaled before the buffer is unmapped so if we want to make sure any transfer operations are finished, we should do it after the unmap call.

Reply:
Done.